setuptools/config/_validate_pyproject/error_reporting.py

```diff
@@ -3,12 +3,21 @@
 import logging
 import os
 import re
+import typing
 from contextlib import contextmanager
 from textwrap import indent, wrap
-from typing import Any, Dict, Iterator, List, Optional, Sequence, Union, cast
+from typing import Any, Dict, Generator, Iterator, List, Optional, Sequence, Union, cast
 
 from .fastjsonschema_exceptions import JsonSchemaValueException
 
+if typing.TYPE_CHECKING:
+    import sys
+
+    if sys.version_info < (3, 11):
+        from typing_extensions import Self
+    else:
+        from typing import Self
```

Comment:
Woops, to be fixed upstream, but this double check is redundant!
```suggestion
if typing.TYPE_CHECKING:
    from typing_extensions import Self
```

Reply:
Not sure if I introduced this one myself, but I think I get why it is there... I think is there this code will allow automatic update (using either ruff or pyupgrade) once Python 3.11 comes?

I agree that is not needed, but I would not worry too much about it.

Reply:
Yeah it's minor. Type-checkers will still understand it and typing_extensions is still not used at runtime.
If it's just an artefact of code generation logic it doesn't hurt, just redundant.